packages/core/src/browser/connection-status-service.ts

```diff
@@ -155,18 +144,32 @@ export class FrontendConnectionStatusService extends AbstractConnectionStatusSer
     }
 
     protected schedulePing(): void {
-        if (this.scheduledPing) {
-            this.clearTimeout(this.scheduledPing);
-        }
+        this.clearTimeout(this.scheduledPing);
         this.scheduledPing = this.setTimeout(async () => {
-            try {
-                await this.pingService.ping();
-                this.updateStatus(true);
-            } catch (e) {
-                this.logger.trace(e);
-            }
+            await this.performPingRequest();
             this.schedulePing();
-        }, this.options.offlineTimeout * 0.8);
+        }, this.options.offlineTimeout);
+    }
+
+    protected async performPingRequest(): Promise<void> {
+        try {
+            await this.pingService.ping();
+            this.updateStatus(true);
+        } catch (e) {
+            this.updateStatus(false);
+            await this.logger.trace(e);
```

Comment:
FYI: This code never happens. Once the WS connection is closed, none of the proxy calls reject but wait until there is a new connection, or forever: https://github.com/eclipse-theia/theia/pull/8803#issue-529643634

I do not mind if you leave it there, but you can simplify with `this.logger.error(e)` as your code does not do what you want to do it. `trace` lavel is useless here.

Reply:
Changed from trace to error.

>Once the WS connection is closed, none of the proxy calls reject but wait until there is a new connection, or forever: #8803 (comment)

AFAIU, this will happened only on browser environment, right?

Reply:
> AFAIU, this will happened only on browser environment, right?

In both. The PR was never merged.